synapse/federation/transport/server/federation.py

```diff
@@ -499,6 +499,11 @@ async def on_PUT(
         result = await self.handler.on_invite_request(
             origin, event, room_version_id=room_version
         )
+
+        # We only store invite_room_state for internal use, so remove it before
+        # returning the event to the remote homeserver.
+        result["event"].get("unsigned", {}).pop("invite_room_state", None)
+
```

Comment:
Will we expose `unsigned.invite_room_state` in other endpoints? (and does that matter?)

Reply:
The field is purposely stripped when serialising an event for clients:

https://github.com/matrix-org/synapse/blob/06df5d4250f54d5a95b0c90bfc9352ec6f02c520/synapse/events/utils.py#L371-L377

which covers the areas where clients could get access to this field. As for federation, it may be included during a backfill... perhaps it's something we should additionally add to `filter_events_for_server`:

https://github.com/matrix-org/synapse/blob/535f8c8f7d64d4058500a5988278fd3026645164/synapse/visibility.py#L534-L623

Reply:
Actually, perhaps it's best if we just stop storing these fields altogether. I've filed https://github.com/matrix-org/synapse/issues/14160 to discuss it.